src/plugins/kibana/server/routes/api/settings/register_delete.js

```diff
@@ -0,0 +1,19 @@
+import Boom from 'boom';
+
+export default function registerDelete(server) {
+  server.route({
+    path: '/api/kibana/settings/{key}',
+    method: 'DELETE',
+    handler: async function (req, reply) {
```

Comment:
This and the other handlers are written as `async` functions but don't use `await` or even return their promises.